api/python/t4/data_transfer.py

```diff
@@ -359,13 +361,18 @@ def copy_object(src, dest, meta, version=None):
             VersionId=version
         )
 
-    s3_client.copy_object(
-        CopySource=src_params,
-        Bucket=dest_bucket,
-        Key=dest_key,
-        Metadata={HELIUM_METADATA: json.dumps(meta)}
-    )
-
+    try:
+        s3_client.copy_object(
+            CopySource=src_params,
+            Bucket=dest_bucket,
+            Key=dest_key,
+            Metadata={HELIUM_METADATA: json.dumps(meta)}
+        )
+    except ClientError as e:
+        # suppress error from copying a file to itself
+        if str(e) == NO_OP_COPY_ERROR_MESSAGE:
```

Comment:
Is this not an EntityAlreadyExists error? 

I'm confused about the specifics of when this arrises. Is this copying and s3 object to itself (same src and dest uri)?  Or is this copying an s3 object to an existing s3 object with identical underlying data (differing src and dest uri)?

Can you add some context to your PR description to make it clearer for reviewing?

Reply:
it's a ClientException, because boto3 exceptions aren't the most helpful. The issue arises when you copy an object to itself without changing anything (metadata, storage class, etc.) I will document this in PR description

Reply:
It seems like we shouldn't make a call to copy an object onto itself, I'm trying to understand when it occurs and why we can't just always skip the s3 api call to copy.

I guess the flow is : pull down an existing manifest with remote s3 keys (not-materialized) and then modify the metadata and publish it back to s3. The copy can't know the metadata has been updated or not just from its current metadata. Is that the correct issue? If so, can we add that to the description.

Reply:
It looks like a more stable way to do this is
```
['Error']['Code']
```
But is the error code is `InvalidRequest` then maybe it is too broad.

https://stackoverflow.com/questions/33068055/boto3-python-and-how-to-handle-errors

Add a todo for now to clean this up. Error messages are generally not stable across api calls and may have i18n issues.